algorithms.py

```diff
@@ -825,6 +890,160 @@ def hudson_recombination_event(self, label, return_heads=False):
+        else:
+            # Get the segment y2 containing the end of the conversion tract
+            y2 = y
+            while y2 is not None and k+tl >= y2.right:
+                y2 = y2.next
+            # process left break
+            if k <= y.left:
+                if x is not None:
+                    x.next = None
+                y.prev = None
+                z = y
+                lhs_tail = x
+            elif k > y.left:
+                z = self.alloc_segment(
+                    k, y.right, y.node, y.population, None, y.next)
+                self.L[label].set_value(z.index, z.right - z.left)
+                if y.next is not None:
+                    y.next.prev = z
+                y.next = None
+                y.right = k
+                self.L[label].increment(y.index, k - z.right)
+                lhs_tail = y
+            # process right break
+            if y2 is not None:
+                if y2.left < k + tl:
+                    z2 = self.alloc_segment(
+                        k + tl, y2.right, y2.node, y2.population, lhs_tail, y2.next)
+                    if lhs_tail is not None:
+                        lhs_tail.next = z2
+                        lhs = lhs_tail.right
+                    else:
+                        print("this should never be possible(2)")
+                        lhs = z2.left
+                    self.L[label].set_value(z2.index, z2.right - lhs)
+                    if y2.next is not None:
+                        y2.next.prev = z2
+                    y2.next = None
+                    y2.right = k + tl
+                    self.L[label].increment(y2.index, k + tl - z2.right)
+                    if z2.prev is None:
+                        z = z2
+                elif y2.left >= k + tl:
+                    lhs_tail.next = y2
+                    y2.prev.next = None
+                    y2.prev = lhs_tail
+                    self.L[label].set_value(y2.index, y2.right - lhs_tail.right)
+        # update population
+        z.label = label
+        self.L[label].set_value(z.index, z.right - z.left - 1)
+        self.P[z.population].add(z, label)
+        # TODO check what needs to be added for full arg
+        ret = None
+        if return_heads:
+            x = lhs_tail
+            # Seek back to the head of the x chain
+            while x.prev is not None:
+                x = x.prev
+            ret = x, z
+        return ret
+
+    def wiuf_geneconversion_left_event(self, label, return_heads=False):
+        """
+        Implements a gene conversion event that started left of a first segment.
+        """
+        self.num_gc_events += 1
+        h = random.uniform(0, self.get_cleft_total(self.tracklength))
+        # Get segment where gene conversion starts from left and length of the individual
+        index, distance = self.find_cleft_individual(h, self.tracklength)
+        y = self.segments[index]
+        # generate tracklength
+        k = y.left + math.floor(1.0 +
```

Comment:
What's the distribution of the value we're adding to y.left? I thought it would be a geometric of some sort.

Reply:
It is a geometrically distributed RV, but conditioned to be smaller than distance. This way the gene conversion will never cover the whole individual from left to right, which would have no effect anyway.